onebusaway-android/src/main/java/org/onebusaway/android/directions/util/ConversionUtils.java

```diff
@@ -0,0 +1,464 @@
+        timeFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+        dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+        CharSequence timeUpdatedString;
+        CharSequence beforeDateString = "", beforeTimeString = "", newDateString = "", timezone = "";
+        SpannableString oldTimeString, oldDateString, newTimeString;
+        calOldTime.setTimeInMillis(oldTime);
+        calNewTime.setTimeInMillis(newTime);
+
+        String noDeviceTimezoneNote = "";
+        if (offsetGMT != TimeZone.getDefault().getOffset(oldTime)) {
+            noDeviceTimezoneNote = "GMT";
+            if (offsetGMT != 0) {
+                noDeviceTimezoneNote += offsetGMT / 3600000;
+            }
+        }
+
+        calOldTime.add(Calendar.MILLISECOND, offsetGMT);
+        calNewTime.add(Calendar.MILLISECOND, offsetGMT);
+
+        if (ConversionUtils.isTomorrow(calNewTime)) {
+            oldDateString = new SpannableString(applicationContext.getResources()
+                    .getString(R.string.time_connector_next_day) + " ");
+        } else {
+            beforeDateString = applicationContext.getResources()
+                    .getString(R.string.time_connector_before_date) + " ";
+            oldDateString = new SpannableString(dateFormat.format(calNewTime.getTime()) + " ");
+        }
+
+        if (calNewTime.get(Calendar.DAY_OF_MONTH) != calOldTime.get(Calendar.DAY_OF_MONTH)) {
+            beforeDateString = applicationContext.getResources()
+                    .getString(R.string.time_connector_before_date) + " ";
+            newDateString = dateFormat.format(calNewTime.getTime()) + " ";
+            oldDateString.setSpan(new StrikethroughSpan(), 0, oldDateString.length() - 1, 0);
+        }
+
+        beforeTimeString = applicationContext.getResources()
+                .getString(R.string.time_connector_before_time) + " ";
+        timezone = noDeviceTimezoneNote;
+
+        int color = 1; /*TODO*/ //getDelayColor(calNewTime.getTimeInMillis() - calOldTime.getTimeInMillis(),
+                //applicationContext);
+        newTimeString = new SpannableString(timeFormat.format(calNewTime.getTime()) + " ");
+        newTimeString.setSpan(new ForegroundColorSpan(color), 0, newTimeString.length(), 0);
+        if (calOldTime.get(Calendar.HOUR_OF_DAY) != calNewTime.get(Calendar.HOUR_OF_DAY)
+                || calOldTime.get(Calendar.MINUTE) != calNewTime.get(Calendar.MINUTE)) {
+            oldTimeString = new SpannableString(timeFormat.format(calOldTime.getTime()) + " ");
+            oldTimeString.setSpan(new StrikethroughSpan(), 0, oldTimeString.length() - 1, 0);
+        } else {
+            oldTimeString = new SpannableString(" ");
+        }
+        timeUpdatedString = TextUtils.concat(beforeDateString, newDateString, oldDateString,
+                beforeTimeString, oldTimeString, newTimeString, timezone);
+        return timeUpdatedString;
+    }
+
+
+    public static int getDelayColor(long delay, Context applicationContext) {

```

Comment:
This method can be removed, and `ArrivalInfo.computeColorFromDeviation()` can be used instead.  Just be aware that that method returns the resource ID, which would still need to be converted to a color via `getResources().getColor(R.color.xxx)` (the below method returns the color integer directly).